src/Tasks/Microsoft.NET.Build.Tasks/ItemUtilities.NuGet.cs

```diff
@@ -11,12 +11,11 @@ internal static partial class ItemUtilities
     {
         public static PackageIdentity GetPackageIdentity(ITaskItem item)
         {
-            string packageName = item.GetMetadata(MetadataKeys.PackageName);
+            string packageName = item.GetMetadata(MetadataKeys.NuGetPackageId);
             string packageVersion = item.GetMetadata(MetadataKeys.PackageVersion);
 
-            if (string.IsNullOrEmpty(packageName) || string.IsNullOrEmpty(packageVersion))
+            if (string.IsNullOrEmpty(packageVersion))
             {
-                packageName = item.GetMetadata(MetadataKeys.NuGetPackageId);
                 packageVersion = item.GetMetadata(MetadataKeys.NuGetPackageVersion);
```

Comment:
Looks like we may have duplication between PackageVersion and NuGetPackageVersion.  We can address that separately though.